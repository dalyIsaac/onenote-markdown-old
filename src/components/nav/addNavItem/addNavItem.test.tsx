import * as enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import { Icon, setIconOptions } from "office-ui-fabric-react";
import * as React from "react";
import AddNavItem, { IPropsAddNavItem } from "./addNavItem";

// Suppress icon warnings.
setIconOptions({
  disableWarnings: true
});

enzyme.configure({ adapter: new Adapter() });

function setUp() {
  const props: IPropsAddNavItem = {
    iconName: "Add",
    onClick: jest.fn(),
    text: "Open notebooks"
  };
  const wrapper = enzyme.mount(<AddNavItem {...props} />);
  return { props, wrapper };
}

describe("Components: AddNavItem", () => {
  test("It should render self and subcomponents", () => {
    const { wrapper, props } = setUp();

    const button = wrapper.find("button").first();
    expect(button.hasClass("addNavItemButton")).toBe(true);

    const divAddNavItemWrapper = wrapper.find("div").first();
    expect(divAddNavItemWrapper.hasClass("addNavItemWrapper")).toBe(true);
    expect(divAddNavItemWrapper.parent().is("button.addNavItemButton")).toBe(
      true
    );

    const icon = wrapper.find(Icon).first();
    expect(icon.hasClass("icon addIcon")).toBe(true);
    expect(icon.parent().is("div.addNavItemWrapper")).toBe(true);

    const label = wrapper.find("label").first();
    expect(label.hasClass("addNavItemLabel")).toBe(true);
    expect(label.text()).toBe(props.text);
    expect(label.parent().is("div.addNavItemWrapper")).toBe(true);
  });
});
