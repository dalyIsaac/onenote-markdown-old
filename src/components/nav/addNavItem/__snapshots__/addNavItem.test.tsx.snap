// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: AddNavItem It should render self and subcomponents 1`] = `
<button
  className="addNavItemButton"
  onClick={[MockFunction]}
>
  <div
    className="addNavItemWrapper"
  >
    <i
      aria-hidden={true}
      className="icon addIcon root-42"
      data-icon-name="Add"
      role="presentation"
    />
    <label
      className="addNavItemLabel"
    >
      Open notebooks
    </label>
  </div>
</button>
`;
